### ISSUE TITLE
tinybmp does not support negative height

### ISSUE THREAD

**Issue by hpux735** [embedded-graphics/tinybmp#8]
- Version of tinybmp in use (if applicable): `0.2.3`

## Description of the problem/feature request/other

Apparently BMP files can have a negative height if the image data is stored top-down, rather than bottom up (as is typical).  https://issues.apache.org/jira/browse/IMAGING-162  The Tinybmp library interprets this as a u32, and predictibly gets upset when the expected large amount of data isn't there.

## Test case (if applicable)

```rust
fn main() -> Result<()> { 
    color_eyre::install()?;

    // #[cfg(feature = "simulator")]
    let mut display: SimulatorDisplay<Rgb888> = SimulatorDisplay::new(Size::new(240, 135));

    let image = Bmp::from_slice(include_bytes!("../resources/rust_pride.bmp")).unwrap();
    let image_raw: ImageRaw<Rgb888> = ImageRaw::new(image.image_data(), image.width(), image.height());
    let object = Image::new(&image_raw, Point::zero());
    object.draw(&mut display);

    let output_settings = OutputSettingsBuilder::new()
        .build();

    Window::new("Hello World", &output_settings).show_static(&display);

    Ok(())
}
```

[rust_pride.bmp.zip](https://github.com/embedded-graphics/tinybmp/files/6527400/rust_pride.bmp.zip)

**Comment by hpux735** [comment]
I've started working on this [here](https://github.com/embedded-graphics/tinybmp/pull/9), but the question I guess I have is whether to flip the lines of the image on ingest or access.  Because "normal" bitmaps are bottom-up, I thought I'd be able to identify where in the existing code that happens and skip it, but I couldn't actually find it.

**Comment by hpux735** [comment]
Ok, work on the MR is now complete, and ready for review.  Let me know if storing that data in the header is overly objectionable, and whether it should be moved somewhere else. 

Thanks!

**Comment by jamwaffles** [comment]
Closed by #9
